File: csharp/coupon/DeleteCouponsByOid.cs
using System;
using System.Collections.Generic;
using System.Reflection;
using com.ultracart.admin.v2.Api;
using com.ultracart.admin.v2.Model;

namespace SdkSample.coupon
{
    public class DeleteCouponsByOid
    {
        
        /// <summary>
        /// Deletes a specific coupon using the UltraCart API
        /// </summary>
        public static void Execute()
        {
            Console.WriteLine("--- " + MethodBase.GetCurrentMethod()?.DeclaringType?.Name + " ---");

            CouponApi couponApi = new CouponApi(Constants.ApiKey);
            string expand = null; // coupons do not have expansions.

            String merchantCode = Guid.NewGuid().ToString("N").Substring(0, 8);
            
            Coupon coupon = new Coupon();
            coupon.MerchantCode = merchantCode; 
            coupon.Description = "Test coupon for DeleteCouponsByCode";
            coupon.AmountOffSubtotal = new CouponAmountOffSubtotal("USD", 0.01m); // one penny discount.

            CouponResponse couponResponse = couponApi.InsertCoupon(coupon, expand);
            coupon = couponResponse.Coupon;

            Console.WriteLine("Created the following temporary coupon:");
            Console.WriteLine($"Coupon OID: {coupon.MerchantCode}");
            Console.WriteLine($"Coupon Type: {coupon.CouponType}");
            Console.WriteLine($"Coupon Description: {coupon.Description}");
            
            // Delete the coupon
            CouponDeletesRequest deleteRequest = new CouponDeletesRequest();
            deleteRequest.CouponOids = new List<int> { coupon.CouponOid };             
            couponApi.DeleteCouponsByCode(deleteRequest);

            Console.WriteLine($"Successfully deleted coupon with merchantCode: {merchantCode}");
        }
    }
}

File: typescript/coupon/DeleteCouponsByOid.ts
import {couponApi} from '../api';
import {
    Coupon,
    CouponAmountOffSubtotal, CouponDeletesRequest,
    CouponResponse, DeleteCouponsByCodeRequest
} from 'ultracart_rest_api_v2_typescript';

export class DeleteCouponByOid {
    /**
     * Deletes a specific coupon using the UltraCart API
     */
    public static async execute(): Promise<void> {
        console.log("--- DeleteCouponByOid ---");

        const expand: string | undefined = undefined; // coupons do not have expansions.

        const merchant_code: string = this.generateGuid().substring(0, 8);

        const coupon: Coupon = {
            merchant_code: merchant_code,
            description: "Test coupon for sdk_sample.coupon.DeleteCoupon",
            amount_off_subtotal: {currency_code: "USD", discount_amount: 0.01} as CouponAmountOffSubtotal
        }; // one penny discount.

        const couponResponse: CouponResponse = await couponApi.insertCoupon({coupon: coupon, expand: expand});
        const createdCoupon: Coupon = couponResponse.coupon!;

        console.log("Created the following temporary coupon:");
        console.log(`Coupon OID: ${createdCoupon.coupon_oid}`);
        console.log(`Coupon Type: ${createdCoupon.coupon_type}`);
        console.log(`Coupon Description: ${createdCoupon.description}`);

        const couponOid: number = createdCoupon.coupon_oid!;
        await couponApi.deleteCouponsByOid({couponDeleteRequest: {coupon_oids: [couponOid]}});

        console.log(`Successfully deleted coupon with ID: ${couponOid}`);
    }

    // Helper method to generate a GUID-like string since TypeScript doesn't have Guid.NewGuid()
    private static generateGuid(): string {
        return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
            const r = Math.random() * 16 | 0;
            const v = c === 'x' ? r : (r & 0x3 | 0x8);
            return v.toString(16);
        }).replace(/-/g, '');
    }
}